// @apache-superset/core/lib/theme/Theme.js
import { jsx as _jsx, jsxs as _jsxs } from "@emotion/react/jsx-runtime";
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
/* eslint-disable react-prefer-function-component/react-prefer-function-component */
// eslint-disable-next-line no-restricted-syntax
import React from 'react';
import { theme as antdThemeImport, ConfigProvider } from 'antd';
import { ThemeProvider, CacheProvider as EmotionCacheProvider, } from '@emotion/react';
import createCache from '@emotion/cache';
import { noop, mergeWith } from 'lodash';
import { GlobalStyles } from './GlobalStyles';
import { allowedAntdTokens, } from './types';
import { normalizeThemeConfig, serializeThemeConfig } from './utils';
export class Theme {
    constructor({ config }) {
        this.SupersetThemeProvider = this.SupersetThemeProvider.bind(this);
        this.setConfig(config || {});
    }
    /**
     * Create a theme from any theme configuration
     * Automatically handles both AntdThemeConfig and SerializableThemeConfig
     * If simple tokens are provided as { token: {...} }, they will be applied with defaults
     * If no config is provided, uses default tokens
     * Dark mode can be set via the algorithm property in the config
     * @param config - The theme configuration
     * @param baseTheme - Optional base theme to apply under the config
     */
    static fromConfig(config, baseTheme) {
        let mergedConfig = config;
        if (baseTheme && config) {
            mergedConfig = mergeWith({}, baseTheme, config, (objValue, srcValue) => Array.isArray(srcValue) ? srcValue : undefined);
            // In Ant Design v5, colorLink derives from colorInfo, not colorPrimary.
            // Currently we expectlinks to follow the brand/primary color. When the user
            // overrides colorPrimary without explicitly setting colorLink, update the
            // merged colorLink so links match the new primary palette.
            if (config.token?.colorPrimary && !config.token?.colorLink) {
                const mToken = mergedConfig?.token;
                if (mToken) {
                    mToken.colorLink = mToken.colorPrimary;
                }
            }
        }
        else if (baseTheme && !config) {
            mergedConfig = baseTheme;
        }
        return new Theme({ config: mergedConfig });
    }
    static getFilteredAntdTheme(antdConfig) {
        // This method generates all antd tokens and filters out the ones not allowed
        // in Superset
        const theme = Theme.getAntdTokens(antdConfig);
        return Object.fromEntries(allowedAntdTokens.map(key => [key, theme[key]]));
    }
    static getAntdTokens(antdConfig) {
        return antdThemeImport.getDesignToken(antdConfig);
    }
    /**
     * Update the theme using any theme configuration
     * Automatically handles both AntdThemeConfig and SerializableThemeConfig
     * Dark mode should be specified via the algorithm property in the config
     */
    setConfig(config) {
        const antdConfig = normalizeThemeConfig(config);
        if (antdConfig.token?.colorPrimary && !antdConfig.token?.colorLink) {
            antdConfig.token.colorLink = antdConfig.token.colorPrimary;
        }
        // First phase: Let Ant Design compute the tokens
        const tokens = Theme.getFilteredAntdTheme(antdConfig);
        // Extract Superset-specific properties from top-level config.
        // These are custom properties that aren't part of Ant Design's token system
        // but need to be passed through to the SupersetTheme for ECharts customization.
        const { echartsOptionsOverrides, echartsOptionsOverridesByChartType } = config;
        // Set the base theme properties
        this.antdConfig = antdConfig;
        this.theme = {
            ...tokens, // First apply Ant Design computed tokens
            ...antdConfig.token, // Then override with our custom tokens
            // Include Superset-specific properties from top-level config
            ...(echartsOptionsOverrides && { echartsOptionsOverrides }),
            ...(echartsOptionsOverridesByChartType && {
                echartsOptionsOverridesByChartType,
            }),
        };
        // Update the providers with the fully formed theme
        this.updateProviders(this.theme, this.antdConfig, createCache({ key: 'superset' }));
    }
    /**
     * Export the current theme as a serializable configuration
     */
    toSerializedConfig() {
        return serializeThemeConfig(this.antdConfig);
    }
    toggleDarkMode(isDark) {
        // Create a new config based on the current one
        const newConfig = { ...this.antdConfig };
        // Determine the new algorithm based on isDark
        const newAlgorithm = isDark
            ? antdThemeImport.darkAlgorithm
            : antdThemeImport.defaultAlgorithm;
        // Handle the case where algorithm is an array
        if (Array.isArray(newConfig.algorithm)) {
            // Filter out any existing dark/default algorithms
            const otherAlgorithms = newConfig.algorithm.filter(alg => alg !== antdThemeImport.darkAlgorithm &&
                alg !== antdThemeImport.defaultAlgorithm);
            // Add the new algorithm to the front of the array
            newConfig.algorithm = [newAlgorithm, ...otherAlgorithms];
        }
        else {
            // Simple case: just replace the algorithm
            newConfig.algorithm = newAlgorithm;
        }
        // Update the theme with the new configuration
        this.setConfig(newConfig);
    }
    json() {
        return JSON.stringify(serializeThemeConfig(this.antdConfig), null, 2);
    }
    updateProviders(theme, antdConfig, emotionCache) {
        noop(theme, antdConfig, emotionCache);
        // Overridden at runtime by SupersetThemeProvider using setThemeState
    }
    SupersetThemeProvider({ children }) {
        if (!this.theme || !this.antdConfig) {
            throw new Error('Theme is not initialized.');
        }
        // eslint-disable-next-line react-hooks/rules-of-hooks
        const [themeState, setThemeState] = React.useState({
            theme: this.theme,
            antdConfig: this.antdConfig,
            emotionCache: createCache({ key: 'superset' }),
        });
        this.updateProviders = (theme, antdConfig, emotionCache) => {
            setThemeState({ theme, antdConfig, emotionCache });
        };
        return (_jsx(EmotionCacheProvider, { value: themeState.emotionCache, children: _jsxs(ThemeProvider, { theme: themeState.theme, children: [_jsx(GlobalStyles, {}), _jsx(ConfigProvider, { theme: themeState.antdConfig, children: children })] }) }));
    }
}
