// @apache-superset/core/lib/theme/types.js
/**
 * Theme algorithms supported by Antd.
 * They can be used individually or in combination.
 * - DEFAULT: Default light theme
 * - DARK: Dark theme
 * - COMPACT: Compact theme (smaller spacing)
 */
export var ThemeAlgorithm;
(function (ThemeAlgorithm) {
    ThemeAlgorithm["DEFAULT"] = "default";
    ThemeAlgorithm["DARK"] = "dark";
    ThemeAlgorithm["COMPACT"] = "compact";
})(ThemeAlgorithm || (ThemeAlgorithm = {}));
/**
 * Represents the current theme mode of the app.
 * It can be one of the following:
 * - DEFAULT: Light theme
 * - DARK: Dark theme
 * - SYSTEM: System theme (auto-detects based on system settings)
 */
export var ThemeMode;
(function (ThemeMode) {
    ThemeMode["DEFAULT"] = "default";
    ThemeMode["DARK"] = "dark";
    ThemeMode["SYSTEM"] = "system";
})(ThemeMode || (ThemeMode = {}));
/**
 * This array is used to define which keys from the full Antd token set
 * we actually allow in the SupersetTheme.
 */
export const allowedAntdTokens = [
    'borderRadius',
    'borderRadiusLG',
    'borderRadiusOuter',
    'borderRadiusSM',
    'borderRadiusXS',
    'boxShadow',
    'boxShadowDrawerLeft',
    'boxShadowDrawerRight',
    'boxShadowDrawerUp',
    'boxShadowPopoverArrow',
    'boxShadowSecondary',
    'boxShadowTabsOverflowBottom',
    'boxShadowTabsOverflowLeft',
    'boxShadowTabsOverflowRight',
    'boxShadowTabsOverflowTop',
    'boxShadowTertiary',
    'colorError',
    'colorErrorActive',
    'colorErrorBg',
    'colorErrorBgActive',
    'colorErrorBgHover',
    'colorErrorBorder',
    'colorErrorBorderHover',
    'colorErrorHover',
    'colorErrorOutline',
    'colorErrorText',
    'colorErrorTextActive',
    'colorErrorTextHover',
    'colorPrimary',
    'colorPrimaryActive',
    'colorPrimaryBg',
    'colorPrimaryBgHover',
    'colorPrimaryBorder',
    'colorPrimaryBorderHover',
    'colorPrimaryHover',
    'colorPrimaryText',
    'colorPrimaryTextActive',
    'colorPrimaryTextHover',
    'colorSuccess',
    'colorSuccessActive',
    'colorSuccessBg',
    'colorSuccessBgHover',
    'colorSuccessBorder',
    'colorSuccessBorderHover',
    'colorSuccessHover',
    'colorSuccessText',
    'colorSuccessTextActive',
    'colorSuccessTextHover',
    'colorBgBase',
    'colorBgBlur',
    'colorBgContainer',
    'colorBgContainerDisabled',
    'colorBgElevated',
    'colorBgLayout',
    'colorBgMask',
    'colorBgSpotlight',
    'colorBgTextActive',
    'colorBgTextHover',
    'colorBorder',
    'colorBorderBg',
    'colorBorderSecondary',
    'colorFill',
    'colorFillAlter',
    'colorFillContent',
    'colorFillContentHover',
    'colorFillQuaternary',
    'colorFillSecondary',
    'colorFillTertiary',
    'colorHighlight',
    'colorIcon',
    'colorIconHover',
    'colorInfo',
    'colorInfoActive',
    'colorInfoBg',
    'colorInfoBgHover',
    'colorInfoBorder',
    'colorInfoBorderHover',
    'colorInfoHover',
    'colorInfoText',
    'colorInfoTextActive',
    'colorInfoTextHover',
    'colorLink',
    'colorLinkActive',
    'colorLinkHover',
    'colorSplit',
    'colorText',
    'colorTextBase',
    'colorTextDescription',
    'colorTextDisabled',
    'colorTextHeading',
    'colorTextLabel',
    'colorTextLightSolid',
    'colorTextPlaceholder',
    'colorTextQuaternary',
    'colorTextSecondary',
    'colorTextTertiary',
    'colorWarning',
    'colorWarningActive',
    'colorWarningBg',
    'colorWarningBgHover',
    'colorWarningBorder',
    'colorWarningBorderHover',
    'colorWarningHover',
    'colorWarningOutline',
    'colorWarningText',
    'colorWarningTextActive',
    'colorWarningTextHover',
    'colorWhite',
    'controlHeight',
    'controlHeightLG',
    'controlHeightSM',
    'controlHeightXS',
    'controlInteractiveSize',
    'controlItemBgActive',
    'controlItemBgActiveDisabled',
    'controlItemBgActiveHover',
    'controlItemBgHover',
    'controlOutline',
    'controlOutlineWidth',
    'controlPaddingHorizontal',
    'controlPaddingHorizontalSM',
    'controlTmpOutline',
    'fontFamily',
    'fontFamilyCode',
    'fontWeightStrong',
    'fontHeight',
    'fontHeightLG',
    'fontHeightSM',
    'fontSize',
    'fontSizeHeading1',
    'fontSizeHeading2',
    'fontSizeHeading3',
    'fontSizeHeading4',
    'fontSizeHeading5',
    'fontSizeIcon',
    'fontSizeLG',
    'fontSizeSM',
    'fontSizeXL',
    'lineHeight',
    'lineHeightHeading1',
    'lineHeightHeading2',
    'lineHeightHeading3',
    'lineHeightHeading4',
    'lineHeightHeading5',
    'lineHeightLG',
    'lineHeightSM',
    'lineType',
    'lineWidth',
    'lineWidthBold',
    'lineWidthFocus',
    'linkDecoration',
    'linkFocusDecoration',
    'linkHoverDecoration',
    'margin',
    'marginLG',
    'marginMD',
    'marginSM',
    'marginXL',
    'marginXS',
    'marginXXL',
    'marginXXS',
    'motion',
    'motionBase',
    'motionDurationFast',
    'motionDurationMid',
    'motionDurationSlow',
    'motionEaseInBack',
    'motionEaseInOut',
    'motionEaseInOutCirc',
    'motionEaseInQuint',
    'motionEaseOut',
    'motionEaseOutBack',
    'motionEaseOutCirc',
    'motionEaseOutQuint',
    'motionUnit',
    'opacityImage',
    'opacityLoading',
    'padding',
    'paddingContentHorizontal',
    'paddingContentHorizontalLG',
    'paddingContentHorizontalSM',
    'paddingContentVertical',
    'paddingContentVerticalLG',
    'paddingContentVerticalSM',
    'paddingLG',
    'paddingMD',
    'paddingSM',
    'paddingXL',
    'paddingXS',
    'paddingXXS',
    'screenLG',
    'screenLGMax',
    'screenLGMin',
    'screenMD',
    'screenMDMax',
    'screenMDMin',
    'screenSM',
    'screenSMMax',
    'screenSMMin',
    'screenXL',
    'screenXLMax',
    'screenXLMin',
    'screenXS',
    'screenXSMax',
    'screenXSMin',
    'screenXXL',
    'screenXXLMin',
    'size',
    'sizeLG',
    'sizeMD',
    'sizeMS',
    'sizePopupArrow',
    'sizeSM',
    'sizeStep',
    'sizeUnit',
    'sizeXL',
    'sizeXS',
    'sizeXXL',
    'sizeXXS',
    'wireframe',
    'zIndexBase',
    'zIndexPopupBase',
];
