// @apache-superset/core/lib/theme/utils/themeUtils.js
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import tinycolor from 'tinycolor2';
import { useTheme as useEmotionTheme } from '@emotion/react';
import { theme as antdTheme } from 'antd';
import { normalizeThemeConfig } from '.';
const fontSizeMap = {
    xs: 'fontSizeXS',
    s: 'fontSizeSM',
    m: 'fontSize',
    l: 'fontSizeLG',
    xl: 'fontSizeXL',
    xxl: 'fontSizeXXL',
};
/**
 * Get font size from theme tokens based on size key
 * @param theme - Theme tokens from useTheme()
 * @param size - Font size key
 * @returns Font size as string
 */
export function getFontSize(theme, size) {
    const key = fontSizeMap[size || 'm'];
    return String(theme[key] || theme.fontSize);
}
/**
 * Check if the current theme is dark mode based on background color
 * @param theme - Theme tokens from useTheme()
 * @returns true if theme is dark, false if light
 */
export function isThemeDark(theme) {
    return tinycolor(theme.colorBgContainer).isDark();
}
/**
 * Get color variants for a given color type from theme tokens
 * @param theme - Theme tokens from useTheme()
 * @param color - Color type (e.g., 'primary', 'error', 'success')
 * @returns ColorVariants object with bg, border, text colors etc.
 */
export function getColorVariants(theme, color) {
    const firstLetterCapped = color.charAt(0).toUpperCase() + color.slice(1);
    if (color === 'default' || color === 'grayscale') {
        const isDark = isThemeDark(theme);
        const flipBrightness = (baseColor) => {
            if (!isDark)
                return baseColor;
            const { r, g, b } = tinycolor(baseColor).toRgb();
            const invertedColor = tinycolor({ r: 255 - r, g: 255 - g, b: 255 - b });
            return invertedColor.toHexString();
        };
        return {
            active: flipBrightness('#222'),
            textActive: flipBrightness('#444'),
            text: flipBrightness('#555'),
            textHover: flipBrightness('#666'),
            hover: flipBrightness('#888'),
            borderHover: flipBrightness('#AAA'),
            border: flipBrightness('#CCC'),
            bgHover: flipBrightness('#DDD'),
            bg: flipBrightness('#F4F4F4'),
        };
    }
    return {
        active: theme[`color${firstLetterCapped}Active`],
        textActive: theme[`color${firstLetterCapped}TextActive`],
        text: theme[`color${firstLetterCapped}Text`],
        textHover: theme[`color${firstLetterCapped}TextHover`],
        hover: theme[`color${firstLetterCapped}Hover`],
        borderHover: theme[`color${firstLetterCapped}BorderHover`],
        border: theme[`color${firstLetterCapped}Border`],
        bgHover: theme[`color${firstLetterCapped}BgHover`],
        bg: theme[`color${firstLetterCapped}Bg`],
    };
}
/**
 * Check if a theme configuration results in a dark theme
 * @param config - The theme configuration to check
 * @returns true if the config results in a dark theme, false otherwise
 */
export function isThemeConfigDark(config) {
    try {
        const normalizedConfig = normalizeThemeConfig(config);
        const themeConfig = antdTheme.getDesignToken(normalizedConfig);
        return tinycolor(themeConfig.colorBgContainer).isDark();
    }
    catch {
        return false;
    }
}
/**
 * Hook to determine if the current theme is dark mode
 * @returns true if theme is dark, false if light
 */
export function useThemeMode() {
    const theme = useEmotionTheme();
    return isThemeDark(theme);
}
